datalad_container/utils.py

```diff
@@ -9,3 +24,128 @@ def get_container_command():
+      If not (matching) container configuration exists, and empty dictionary
+      is returned.
+    """
+    var_prefix = 'datalad.containers.'
+
+    containers = {}
+    # all info is in the dataset config!
+    for var, value in ds.config.items():
+        if not var.startswith(var_prefix):
+            # not an interesting variable
+            continue
+        var_comps = var.split('.')
+        # container name is the 3rd after 'datalad'.'container'.
+        cname = var_comps[2]
+        if name and name != cname:
+            # we are looking for a specific container's configuration
+            # and this is not it
+            continue
+        # reconstruct config item name, anything after
+        # datalad.containers.<name>.
+        ccfgname = '.'.join(var_comps[3:])
+        if not ccfgname:
+            continue
+
+        if ccfgname == 'image':
+            # run image path normalization to get a relative path
+            # in platform conventions, regardless of the input.
+            # for now we report a str, because the rest of the code
+            # is not using pathlib
+            value = str(_normalize_image_path(value, ds))
+
+        cinfo = containers.get(cname, {})
+        cinfo[ccfgname] = value
+
+        containers[cname] = cinfo
+
+    return containers if name is None else containers.get(name, {})
+
+
+def _normalize_image_path(path: str, ds: Dataset) -> str:
+    """Helper to standardize container image path handling
+
+    Previously, container configuration would contain platform-paths
+    for container image location (e.g., windows paths when added on
+    windows, POSIX paths elsewhere). This made cross-platform reuse
+    impossible out-of-the box, but it also means that such dataset
+    are out there in unknown numbers.
+
+    This helper inspects an image path READ FROM CONFIG(!) and ensures
+    that is matches platform conventions (because all other arguments)
+    also come in platform conventions. This enables standardizing
+    the storage conventions to be POSIX-only (for the future).
+
+    Parameters
+    ----------
+    path: str
+      A str-path, as read from the configuration, matching its conventions
+      (relative path, pointing to a container image relative to the
+      dataset's root.
+    ds: Dataset
+      This dataset's base path is used as a reference for resolving
+      the relative image path to an absolute location on the file system.
+
+    Returns
+    -------
+    PurePath
+      Relative path in platform conventions
+    """
+    # we only need to act differently, when an incoming path is
+    # windows. This is not possible to say with 100% confidence,
+    # because a POSIX path can also contain a backslash. We support
+    # a few standard cases where we CAN tell
+    pathobj = None
+    if '\\' not in path:
+        # no windows pathsep, no problem
+        pathobj = PurePosixPath(path)
+    elif path.startswith('.datalad\\environments\\'):
```

Comment:
are you sure you don't need a
```suggestion
    elif path.startswith(r'.datalad\\environments\\'):
```
?  IIRC you showed that it really stores `\\` in the path in the config, or it converts them to a single `\` upon loading?  (in the [test above](https://github.com/datalad/datalad-container/pull/243/files#diff-c9a6a0cde74f0c35ded765fc396d537f9485fe47645c4e00698ab951138042d6R408) you also set to a single `\` since lacking `r""` so I wondered to double check if that corresponds to actual target to fix)

Reply:
Oh man, wtf!? you are absolutely correct!! My brain did an internal `print()` all these days. I just went back and confirmed again that the report in https://github.com/datalad/datalad-container/issues/224 is correct and we do have `\\` in the config files generated on windows, not `\`

:facepalm: